### ISSUE TITLE
Helm chart Capabilities.APIVersions does not support v1 API

### ISSUE THREAD

**Issue by prune998** [coreos/prometheus-operator#1714]
**What did you do?**
I build a deployment file from the Helm chart using : 
```
helm template helm/prometheus-operator --name prometheus-operator --namespace monitoring -f operator_values.yml > operator_generated.yml
```

**What did you expect to see?**
a valid Manifest to deploy the Operator

**What did you see instead? Under which circumstances?**
Some parts of the Manifest does not include the `apivVersion`: 

```
# Source: prometheus-operator/templates/clusterrole.yaml

kind: ClusterRole
metadata:
  labels:
    app: prometheus-operator
    ...
```

Looking at the file `prometheus-operator/templates/clusterrole.yaml` I see : 
```
{{- if .Capabilities.APIVersions.Has "rbac.authorization.k8s.io/v1beta1" }}
apiVersion: rbac.authorization.k8s.io/v1beta1
{{- else if .Capabilities.APIVersions.Has "rbac.authorization.k8s.io/v1alpha1" }}
apiVersion: rbac.authorization.k8s.io/v1alpha1
{{- end }}
```

I did print the `.Capabilities.APIVersions` variable as a Label, and here is the result : 
```
api: map[v1:{}]
```

So that makes the Helm chart incompatible with Kubernetes 1.10+
That also make the chart really hard to maintain and will require more and more `if`/`else` when the API changes

**Environment**

* Prometheus Operator version:

clone from master
`commit 5318247b242823e6f45266f678855f0eef215f05 (HEAD -> master, origin/master, origin/HEAD)`

* Kubernetes version information:

```
kubectl version
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.0", GitCommit:"91e7b4fd31fcd3d5f436da26c980becec37ceefe", GitTreeState:"clean", BuildDate:"2018-06-27T22:30:22Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.5-gke.3", GitCommit:"6265b9797fc8680c8395abeab12c1e3bad14069a", GitTreeState:"clean", BuildDate:"2018-07-19T23:02:51Z", GoVersion:"go1.9.3b4", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: 

	1.10.5-gke.3

**Comment by kingdonb** [comment]
> I did print the .Capabilities.APIVersions variable as a Label, and here is the result :
> 
> api: map[v1:{}]

Is that really what you found in `.Capabilities.APIVersions`?  If so, I think that's very surprising ... we have been running into similar issues and our solution was to parse Minor as an int and decide based on what API versions we are likely to see (<1.6 has alpha, if any RBAC at all...), well I'm not sure but I think this method also fails because "10+" is not parseable as an int.

The issue will be fixed by #1729 I think, but if you run into a cluster that does have both `rbac.authorization.k8s.io/v1beta1:{}` and `rbac.authorization.k8s.io/v1:{}` in ther APIVersions, you're going to choose `v1beta1` and I think that might be less than optimal...

Does anyone know when and where the decision to hollow out `.Capabilities.APIVersions` was made and why?  What other alternative is there for detecting things like this?

**Comment by prune998** [comment]
its not empty. It's displaying `v1` only as only the key needs to be present. 
I think that older k8s should have displayed `[v1alpha:{},v1beta:{}]`

**Comment by gianrubio** [comment]
Fixed #1729
